cilium-dbg/cmd/envoy_adminclient.go

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"path/filepath"
+
+	"github.com/cilium/cilium/pkg/safeio"
+)
+
+const envoySocketDirPath = "/var/run/cilium/envoy/sockets/"
+
+type envoyAdminClient struct {
+	adminURL string
+	unixPath string
+}
+
+func newEnvoyAdminClient() *envoyAdminClient {
+	return &envoyAdminClient{
+		// Needs to be provided to envoy (received as ':authority') - even though we Dial to a Unix domain socket.
+		adminURL: fmt.Sprintf("http://%s/", "envoy-admin"),
+		unixPath: filepath.Join(envoySocketDirPath, "admin.sock"),
+	}
+}
+
+func (a *envoyAdminClient) get(path string) (string, error) {
+	// Use a custom dialer to use a Unix domain socket for a HTTP connection.
+	var conn net.Conn
+	var err error
+	client := &http.Client{
+		Transport: &http.Transport{
+			DialContext: func(_ context.Context, _, _ string) (net.Conn, error) {
+				conn, err = net.Dial("unix", a.unixPath)
+				return conn, err
+			},
+		},
+	}
+
+	resp, err := client.Get(fmt.Sprintf("%s%s", a.adminURL, path))
+	if err != nil {
+		return "", fmt.Errorf("failed to call %q endpoint: %w", path, err)
```

Comment:
probably just parse and validate url as we are accepting user input

```diff
diff --git a/cilium-dbg/cmd/envoy_adminclient.go b/cilium-dbg/cmd/envoy_adminclient.go
index 1976d2cbaa4..9b2560da2d4 100644
--- a/cilium-dbg/cmd/envoy_adminclient.go
+++ b/cilium-dbg/cmd/envoy_adminclient.go
@@ -8,6 +8,7 @@ import (
 	"fmt"
 	"net"
 	"net/http"
+	"net/url"
 	"path/filepath"
 
 	"github.com/cilium/cilium/pkg/safeio"
@@ -41,7 +42,12 @@ func (a *envoyAdminClient) get(path string) (string, error) {
 		},
 	}
 
-	resp, err := client.Get(fmt.Sprintf("%s%s", a.adminURL, path))
+	u, err := url.Parse(fmt.Sprintf("%s%s", a.adminURL, path))
+	if err != nil {
+		return "", fmt.Errorf("failed to parse URL: %w", err)
+	}
+
+	resp, err := client.Get(u.String())
 	if err != nil {
 		return "", fmt.Errorf("failed to call %q endpoint: %w", path, err)
 	}
```

Reply:
Thanks, I updated the PR accordingly!